File: algo_problems/q700+/q800.java
class Solution {
    public String similarRGB(String color) {
        return "#" + f(color.substring(1, 3)) + f(color.substring(3, 5)) + f(color.substring(5, 7));
    }

    public String f(String comp){
    	int v = Integer.parseInt(comp, 16);
    	v = v / 17 + (v % 17 > 8 ? 1 : 0);
    	return String.format("%02x", 17* v);
    }
}

File: algo_problems/q700+/q800.py
class Solution:
    def similarRGB(self, color):
        """
        :type color: str
        :rtype: str
        """
        result = '#'
        for i in range(3):
            s = int(color[i * 2 + 1: i * 2 + 3], 16)
            d = (s // 17) * 17
            if d + 17 - s < s - d:
                result += hex(d + 17)[2:]
            else:
                t = hex(d)[2:]
                result += t if len(t) == 2 else t * 2
        return result


if __name__ == '__main__':
    print(Solution().similarRGB('#001122'))
    print(Solution().similarRGB('#334455'))
    print(Solution().similarRGB('#F03456'))
